smdebug/core/hook.py

```diff
@@ -570,7 +570,9 @@ def _write_scalars(self):
             write_event = scalar_obj.write_event
             if self.metrics_writer and sm_metric:
                 self.metrics_writer.log_metric(
-                    scalar_name, scalar_val, iteration_number=self.mode_steps[self.mode]
+                    scalar_name + "_" + self.mode.name,
```

Comment:
If mode is global, the step number needs to be self.step
That's a separate issue currently it looks like